### PR TITLE
Misc class based syntax clean up

### PR DESCRIPTION
**Pull request by bdemann**

### CODE REVIEW

File: src/lib/stable/ic_apis/caller.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { Principal } from '../';
  */
 export function caller(): Principal {
     if (globalThis._azleIc === undefined) {
-        return undefined as any;
+        return Principal.fromHex('0x04');
```
Comment (lastmjs) [inline src/lib/stable/ic_apis/caller.ts:9]: Probably wrong

File: src/lib/stable/ic_apis/id.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ import { Principal } from '..';
  */
 export function id(): Principal {
     if (globalThis._azleIc === undefined) {
-        return undefined as any;
+        return Principal.fromHex('0x04');
```
Comment (lastmjs) [inline src/lib/stable/ic_apis/id.ts:9]: Probably not the anonymous principal

File: tests/end_to_end/candid_rpc/class_syntax/notify_raw/src/canister1/index.ts [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
             Principal.fromText(
                 process.env.CANISTER2_PRINCIPAL ??
                     trap('process.env.CANISTER2_PRINCIPAL is undefined')
             ),
             'receiveNotification',
-            Uint8Array.from(candidEncode('()')),
-            0n
+            {
+                raw: Uint8Array.from(candidEncode('()')),
+                payment: 0n
```
Comment (lastmjs) [inline tests/end_to_end/candid_rpc/class_syntax/notify_raw/src/canister1/index.ts:14]: Is the payment sent with notify raw?

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: examples/composite_queries/test/tests.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ export function getTests(canister1: ActorSubclass<_SERVICE>): Test {
             expect(result).toBe('Hello from Canister 2');
         });
 
-        it('performs a simple manual query', async () => {
+        it.only('performs a simple manual query', async () => {
```
[the remark this answers is not in the document]
Comment (bdemann) [reply examples/composite_queries/test/tests.ts]: done

File: src/compiler/generate_candid_and_canister_methods.ts
[the remark this answers is not in the document]
Comment (bdemann) [reply src/compiler/generate_candid_and_canister_methods.ts:1]: done

File: src/lib/ic/call.ts [hunk 5]
```diff
@@ -19,36 +17,22 @@ export function call<T extends (...args: any[]) => any>(
...
     config?: {
         args?: ArgsType<T>;
-        cycles?: nat64;
-        cycles128?: nat;
+        cycles?: nat;
     }
 ): ReturnTypeOf<T> {
     if (globalThis._azleIc === undefined) {
-        return undefined as any;
+        return {} as any;
```
[the remark this answers is not in the document]
Comment (bdemann) [reply src/lib/ic/call.ts:24]: done

File: src/lib/stable/ic_apis/call.ts [hunk 6]
```diff
@@ -16,7 +16,7 @@ export async function call(
     // TODO this should use a Result remember
     return new Promise((resolve, reject) => {
         if (globalThis._azleIc === undefined) {
-            return undefined as any;
+            return {};
```
[the remark this answers is not in the document]
Comment (bdemann) [reply src/lib/stable/ic_apis/call.ts:19]: done

File: examples/stable_memory/test/tests.ts
[the remark this answers is not in the document]
Comment (bdemann) [reply examples/stable_memory/test/tests.ts:1]: done

[the remark this answers is not in the document]
Comment (bdemann) [reply examples/stable_memory/test/tests.ts:1]: duplicate

File: tests/end_to_end/candid_rpc/class_syntax/manual_reply/src/index.ts
Comment (lastmjs) [inline tests/end_to_end/candid_rpc/class_syntax/manual_reply/src/index.ts:1]: Figure out reserved, we should be able to pass anything in, and I think it should be undefined

**Review by lastmjs — CHANGES_REQUESTED** [review]

**Review by lastmjs — CHANGES_REQUESTED** [review]
Let's use `candidType` and `idlType` convention everywhere

**Review by lastmjs — DISMISSED** [review]

**Pull request merged**